docs: Add quick-start guide and example.

kirkrodrigues opened the pull request:
<!--
Set the PR title to a meaningful commit message that:
- follows the Conventional Commits specification (https://www.conventionalcommits.org).
- is in imperative form.

Example:
fix: Don't add implicit wildcards ('*') at the beginning and the end of a query (fixes #390).
-->

# Description
<!-- Describe what this request will change/fix and provide any details necessary for reviewers. -->

This PR adds a quick-start guide for writing a basic task and running it on Spider. In future PRs, we will explain how to write complex tasks (e.g., tasks joined together into task graph) as well as how to use Spider's data and key-value stores.

NOTE: This should not be merged until the scheduler and worker can be run as standalone processes.

# Validation performed
<!-- Describe what tests and validation you performed on the change. -->

* Validated the output renders correctly on GitHub.
* Validated the links are correct.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive quick-start guide for users to effectively utilize the Spider distributed system.
	- Added a new task function, `sum`, for calculating the sum of two integers within the Spider framework.
	- Implemented a client application with command line argument parsing, job submission, and result validation.
- **Documentation**
	- Added a new documentation file detailing steps for writing tasks, building shared libraries, and setting up a Spider cluster, including prerequisites and example tasks.
- **Chores**
	- Enhanced linting tasks to include additional source directories and files for improved code quality checks.
	- Updated task configurations to support building examples alongside the main project.
	- Introduced a new clang-format configuration for C++ code formatting in the examples directory.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kirkrodrigues commented:
@coderabbitai review

sitaowang1998 commented on docs/quick-start.md:
````diff
@@ -0,0 +1,324 @@
+
+To build the scheduler, run:
+
+```shell
+cmake -S spider -B spider/build
+cmake --build spider/build --parallel $(nproc) --target spider_scheduler
+```
````
Why build spider inside a separate build directory again? The root level `CMakeLists.txt` includes `add_directory(spider)`, which can be used to spider.

kirkrodrigues commented:
@sitaowang1998 Addressed your comments by moving the quick-start source code into the `examples` directory. It seems like there's a few issues with the client library's CMake set up since building the quick-start example doesn't work. One issue is that the header files for the client library need to be specified using `HEADERS` `FILE_SET`. I think the other is that spdlog is being included in one of the public headers. Can you take a look?

sitaowang1998 commented:
> @sitaowang1998 Addressed your comments by moving the quick-start source code into the `examples` directory. It seems like there's a few issues with the client library's CMake set up since building the quick-start example doesn't work. One issue is that the header files for the client library need to be specified using a `HEADERS` `FILE_SET`. I think the other is that spdlog is being included in one of the public headers. Can you take a look?

Open new pr #46 to solve the cmake and header issue.

sitaowang1998 commented:
@coderabbitai review
